### DISCUSSION TITLE
[python-v2] Question about the "Parameters" argument in generic_input_binding

### DISCUSSION THREAD

**Discussion by frouhi** [Azure/azure-functions-sql-extension#924] [Q&A]
I've been using the generic_input_binding decorator, and it has been pretty neat. One thing I was trying to do, and haven't managed to figure out is about the "Parameters" argument:
https://github.com/Azure/azure-functions-sql-extension/blob/main/samples/samples-python-v2/function_app.py#L22

currently, it can get an argument from the route URL, but is there any way to have some more complex logic there (say pass a function that gets the header, extracts some information from a key in the header, and use that as the parameter)? Or maybe use some form parameters instead of route arguments?

this functionality is pretty nice, but since this is one of the few examples I've managed to find, I've been wondering if there are more advanced things we can do with this parameter other than the url route argument.

**Comment by lucyzhang929** [comment]
Hi @frouhi, you can also pass in parameters using the app settings. Here is an example in Python v1: https://github.com/Azure/azure-functions-sql-extension/blob/1e130fc270980aa37921d89457dbda833e723cb9/samples/samples-python/GetProductsStoredProcedureFromAppSetting/function.json#L24C28-L24C28

The value for %ProductCost% is defined here: https://github.com/Azure/azure-functions-sql-extension/blob/1e130fc270980aa37921d89457dbda833e723cb9/samples/samples-python/local.settings.json#L8C2-L8C23.
